platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/filter/DefaultTestFilter.java

```diff
@@ -16,54 +16,48 @@
 package org.gradle.api.internal.tasks.testing.filter;
 
 import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.SetProperty;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.testing.TestFilter;
+import org.gradle.internal.instrumentation.api.annotations.ReplacesEagerProperty;
+import org.gradle.internal.instrumentation.api.annotations.ReplacesEagerProperty.BinaryCompatibility;
 import org.gradle.internal.scan.UsedByScanPlugin;
 
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.Set;
+import java.util.LinkedHashSet;
 
 @UsedByScanPlugin("test-retry")
-public class DefaultTestFilter implements TestFilter {
+public abstract class DefaultTestFilter implements TestFilter {
```

Comment:
🤔 I wonder if going forward it would make sense to have the implementation of stuff be written in Kotlin, even if the APIs stay Java-only... Doing so would allow us to use a lot of the simpler syntax.

Probably more importantly we'd be writing more similar code to what our users will write more of the time...

Reply:
It would definitely be shorter 🤔